src/wallet/test/coinselector_tests.cpp

```diff
@@ -993,7 +994,7 @@ BOOST_AUTO_TEST_CASE(waste_test)
         const CAmount target_waste2{-2 * large_fee_diff + change_cost}; // = (2 * fee) - (2 * (fee + large_fee_diff)) + change_cost
```

Comment:
nit: do you want to add assert for the comment above? It's not obvious that the statement holds given variable declarations are hundreds of lines above.

the comment:
"Negative waste when the long term fee is greater than the current fee and change_cost < - (inputs * (fee - long_term_fee))"